pkg/bitcoin/address_test.go

```diff
@@ -1,167 +1,60 @@
-			}
-
-			if !reflect.DeepEqual(got, tt.want) {
-				t.Errorf("getChainParams() got = %v, want %v", got, tt.want)
-			}
-		})
-	}
-}
-
 func TestService_ValidateAddress(t *testing.T) {
 	tests := []struct {
-		name    string
-		request *pb.ValidateAddressRequest
-		want    *pb.ValidateAddressResponse
-		wantErr *status.Status
+		name        string
+		address     string
+		chainParams ChainParams
+		want        string
+		wantErr     error
 	}{
 		{
-			name: "mainnet P2PKH valid",
-			request: &pb.ValidateAddressRequest{
-				Address:     "1MirQ9bwyQcGVJPwKUgapu5ouK2E2Ey4gX",
-				ChainParams: mainnetChainParamsProto,
-			},
-			want: &pb.ValidateAddressResponse{
-				Address: "1MirQ9bwyQcGVJPwKUgapu5ouK2E2Ey4gX",
-				IsValid: true,
-			},
-			wantErr: nil,
+			name:        "mainnet P2PKH valid",
+			address:     "1MirQ9bwyQcGVJPwKUgapu5ouK2E2Ey4gX",
+			chainParams: Mainnet,
+			want:        "1MirQ9bwyQcGVJPwKUgapu5ouK2E2Ey4gX",
 		},
 		{
-			name: "mainnet P2WPKH invalid checksum",
-			request: &pb.ValidateAddressRequest{
-				Address:     "bc1qw508d6qejxtdg4y5r3zarvary0c5xw7kv8f3t5",
-				ChainParams: mainnetChainParamsProto,
-			},
-			want: &pb.ValidateAddressResponse{
-				Address:       "bc1qw508d6qejxtdg4y5r3zarvary0c5xw7kv8f3t5",
-				IsValid:       false,
-				InvalidReason: "checksum failed. Expected v8f3t4, got v8f3t5.",
-			},
-			wantErr: nil,
-		},
-		{
-			name: "testnet3 P2WPKH invalid mixed case",
-			request: &pb.ValidateAddressRequest{
-				Address:     "tb1qrp33g0q5c5txsp9arysrx4k6zdkfs4nce4xj0gdcccefvpysxf3q0sL5k7",
-				ChainParams: testnet3ChainParamsProto,
-			},
-			want: &pb.ValidateAddressResponse{
-				Address:       "tb1qrp33g0q5c5txsp9arysrx4k6zdkfs4nce4xj0gdcccefvpysxf3q0sL5k7",
-				IsValid:       false,
-				InvalidReason: "string not all lowercase or all uppercase",
-			},
-			wantErr: nil,
+			name:        "mainnet P2WPKH invalid checksum",
+			address:     "bc1qw508d6qejxtdg4y5r3zarvary0c5xw7kv8f3t5",
+			chainParams: Mainnet,
+			wantErr:     errors.New("checksum failed. Expected v8f3t4, got v8f3t5."),
 		},
 		{
-			name: "get unknown chain params",
-			request: &pb.ValidateAddressRequest{
-				Address:     "1MirQ9bwyQcGVJPwKUgapu5ouK2E2Ey4gX",
-				ChainParams: invalidChainParamsProto,
-			},
-			want:    nil,
-			wantErr: status.New(codes.InvalidArgument, ErrUnknownNetwork("99999").Error()),
+			name:        "testnet3 P2WPKH invalid mixed case",
+			address:     "tb1qrp33g0q5c5txsp9arysrx4k6zdkfs4nce4xj0gdcccefvpysxf3q0sL5k7",
+			chainParams: Testnet3,
+			wantErr:     errors.New("string not all lowercase or all uppercase"),
 		},
 	}
 
-	ctx := context.Background()
 	s := &Service{}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			got, err := s.ValidateAddress(ctx, tt.request)
-			grpcErr := status.Convert(err)
+			got, err := s.ValidateAddress(tt.address, tt.chainParams)
+
+			if err != nil && tt.wantErr == nil {
+				t.Errorf("unexpected error in ValidateAddress(): %v", err)
```

Comment:
You can oneline these 2 lines by: `t.Fatalf("unexpected error in ValidateAddress(): %v", err)`